Adding LTR for MI to .NET SDK

xaliciayang opened the pull request:
Adding clients + test file for Managed Instance Long Term Retention Tests

isra-fel reviewed (changes requested):
Hi @xaliciayang , it failed to build because some of the models could not be found, e.g. `ManagedInstanceLongTermRetentionBackup`. Did you forget to check in them?

Besides, please
- add link(s) to the swagger spec review in this PR's description
- check in the metadata txt of your module under `/eng/mgmt/mgmtmetadata` 

Detailed process guide: https://github.com/Azure/adx-documentation-pr/blob/master/engineering/adx_netsdk_process.md#generating-net-sdk

xaliciayang commented:
Added the missing models (also added many unrelated files in attempt to sync local repo). Will remove out-of-scope files

Here is the swagger PR: https://github.com/Azure/azure-rest-api-specs/pull/8266

pixia commented on sdk/sqlmanagement/Microsoft.Azure.Management.SqlManagement/src/Generated/IManagedDatabasesOperations.cs:
```diff
@@ -174,6 +174,33 @@ public partial interface IManagedDatabasesOperations
+        /// <exception cref="Microsoft.Rest.ValidationException">
+        /// Thrown when a required parameter is null
+        /// </exception>
+        Task<AzureOperationResponse<IPage<ManagedDatabase>>> ListInaccessibleByInstanceWithHttpMessagesAsync(string resourceGroupName, string managedInstanceName, Dictionary<string, List<string>> customHeaders = null, CancellationToken cancellationToken = default(CancellationToken));
```
this is not our change

pixia commented on sdk/sqlmanagement/Microsoft.Azure.Management.SqlManagement/src/Generated/IManagedDatabasesOperations.cs:
```diff
@@ -348,5 +375,27 @@ public partial interface IManagedDatabasesOperations
+        /// <exception cref="Microsoft.Rest.ValidationException">
+        /// Thrown when a required parameter is null
+        /// </exception>
+        Task<AzureOperationResponse<IPage<ManagedDatabase>>> ListInaccessibleByInstanceNextWithHttpMessagesAsync(string nextPageLink, Dictionary<string, List<string>> customHeaders = null, CancellationToken cancellationToken = default(CancellationToken));
```
not our change

pixia commented on sdk/sqlmanagement/Microsoft.Azure.Management.SqlManagement/tests/SessionRecords/ManagedInstanceLongTermRetentionTests/TestManagedInstanceLongTermRetentionPolicies.json:
```diff
@@ -0,0 +1,828 @@
+          "no-cache"
+        ],
+        "Location": [
+          "https://api-dogfood.resources.windows-int.net/subscriptions/8cfb8b62-bcd6-4713-89ad-18097f75cc5b/providers/Microsoft.Sql/locations/southeastasia/managedDatabaseOperationResults/1dd83660-8f4f-4079-952d-2fb139812065?api-version=2019-06-01-preview"
```
are we ok to expose test endpoint?

jaredmoo commented on sdk/sqlmanagement/Microsoft.Azure.Management.SqlManagement/src/Generated/Models/DatabaseState6.cs:
```diff
@@ -0,0 +1,23 @@
+    /// <summary>
+    /// Defines values for DatabaseState6.
+    /// </summary>
+    public static class DatabaseState6
```
Autorest bug??

xaliciayang committed:
adding LTR MI tests for net

xaliciayang committed:
adding test session records

xaliciayang committed:
adding generated files

xaliciayang committed:
adding another generated file

xaliciayang commented on sdk/sqlmanagement/Microsoft.Azure.Management.SqlManagement/src/Generated/SdkInfo_SqlManagementClient.cs:
```diff
@@ -87,7 +89,7 @@ internal static partial class SdkInfo
                 new Tuple<string, string, string>("Sql", "RestorePoints", "2017-03-01-preview"),
                 new Tuple<string, string, string>("Sql", "SensitivityLabels", "2017-03-01-preview"),
                 new Tuple<string, string, string>("Sql", "ServerAutomaticTuning", "2017-03-01-preview"),
-                new Tuple<string, string, string>("Sql", "ServerAzureADAdministrators", "2014-04-01"),
```
@amolagar should I revert this and leave it to you?

pixia commented on sdk/sqlmanagement/Microsoft.Azure.Management.SqlManagement/src/Generated/SqlManagementClient.cs:
```diff
@@ -610,33 +607,6 @@ public SqlManagementClient(ServiceClientCredentials credentials, params Delegati
-        /// <exception cref="System.ArgumentNullException">
-        /// Thrown when a required parameter is null
-        /// </exception>
-        public SqlManagementClient(ServiceClientCredentials credentials, HttpClient httpClient, bool disposeHttpClient) : this(httpClient, disposeHttpClient)
```
why is this removed?

pixia commented on sdk/sqlmanagement/Microsoft.Azure.Management.SqlManagement/src/Generated/SqlManagementClient.cs:
```diff
@@ -483,37 +468,49 @@ public partial class SqlManagementClient : ServiceClient<SqlManagementClient>, I
-        /// </param>
-        /// <param name='disposeHttpClient'>
-        /// True: will dispose the provided httpClient on calling SqlManagementClient.Dispose(). False: will not dispose provided httpClient</param>
-        protected SqlManagementClient(HttpClient httpClient, bool disposeHttpClient) : base(httpClient, disposeHttpClient)
```
why is this removed?

xaliciayang committed:
fixing things that should not have been updated

xaliciayang committed:
additional fix; for API that is not ready

pixia commented on sdk/sqlmanagement/Microsoft.Azure.Management.SqlManagement/src/Generated/SdkInfo_SqlManagementClient.cs:
```diff
@@ -110,19 +112,10 @@ internal static partial class SdkInfo
                 new Tuple<string, string, string>("Sql", "Usages", "2018-06-01-preview"),
                 new Tuple<string, string, string>("Sql", "VirtualClusters", "2015-05-01-preview"),
                 new Tuple<string, string, string>("Sql", "VirtualNetworkRules", "2015-05-01-preview"),
+                new Tuple<string, string, string>("Sql", "WorkloadClassifiers", "2019-06-01-preview"),
```
remove workloadClassifiers

xaliciayang committed:
making sure versions are correct in SdkInfo

xaliciayang committed:
add updated test recordings; add skip tags to some tests

xaliciayang committed:
fix a version

xaliciayang committed:
session records

xaliciayang committed:
reset serverazuread version

xaliciayang committed:
Merge branch 'ayang_net' of https://github.com/xaliciayang/azure-sdk-for-net into ayang_net

xaliciayang committed:
add databasestate files

xaliciayang committed:
make tests and records consistent

xaliciayang committed:
other required files

xaliciayang committed:
rearranging

isra-fel commented on sdk/sqlmanagement/Microsoft.Azure.Management.SqlManagement/src/Microsoft.Azure.Management.Sql.csproj:
```diff
@@ -12,9 +12,11 @@
 			<![CDATA[
 New features:
-- Added support for Azure SQL Server Azure Active Directory administrator API version 2019-06-01-preview.
+- Added support for Azure SQL Server Azure Active Directory administrator API` version 2019-06-01-preview.
```
nit:
```suggestion
- Added support for Azure SQL Server Azure Active Directory administrator API version 2019-06-01-preview.
```

isra-fel committed:
Update sdk/sqlmanagement/Microsoft.Azure.Management.SqlManagement/src/Microsoft.Azure.Management.Sql.csproj

xaliciayang committed:
bump version

xaliciayang committed:
Merge branch 'ayang_net' of https://github.com/xaliciayang/azure-sdk-for-net into ayang_net
